include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce.h

```diff
@@ -289,63 +319,63 @@ __parallel_transform_reduce(_ExecutionPolicy&& __exec, _ReduceOp __reduce_op, _T
-                ::std::forward<_ExecutionPolicy>(__exec), __n, __reduce_op, __transform_op, __init,
-                ::std::forward<_Ranges>(__rngs)...);
+            return __parallel_transform_reduce_small_impl<512, 2, _Tp>(::std::forward<_ExecutionPolicy>(__exec), __n,
+                                                                       __reduce_op, __transform_op, __init,
+                                                                       ::std::forward<_Ranges>(__rngs)...);
         }
         else if (__n <= 2048)
         {
-            return __parallel_transform_reduce_small_submitter<512, 4, _Tp, _ReduceOp, _TransformOp>::submit(
-                ::std::forward<_ExecutionPolicy>(__exec), __n, __reduce_op, __transform_op, __init,
-                ::std::forward<_Ranges>(__rngs)...);
+            return __parallel_transform_reduce_small_impl<512, 4, _Tp>(::std::forward<_ExecutionPolicy>(__exec), __n,
+                                                                       __reduce_op, __transform_op, __init,
+                                                                       ::std::forward<_Ranges>(__rngs)...);
         }
         else if (__n <= 4096)
         {
-            return __parallel_transform_reduce_small_submitter<512, 8, _Tp, _ReduceOp, _TransformOp>::submit(
-                ::std::forward<_ExecutionPolicy>(__exec), __n, __reduce_op, __transform_op, __init,
-                ::std::forward<_Ranges>(__rngs)...);
+            return __parallel_transform_reduce_small_impl<512, 8, _Tp>(::std::forward<_ExecutionPolicy>(__exec), __n,
+                                                                       __reduce_op, __transform_op, __init,
+                                                                       ::std::forward<_Ranges>(__rngs)...);
         }
         else if (__n <= 8192)
         {
-            return __parallel_transform_reduce_small_submitter<512, 16, _Tp, _ReduceOp, _TransformOp>::submit(
-                ::std::forward<_ExecutionPolicy>(__exec), __n, __reduce_op, __transform_op, __init,
-                ::std::forward<_Ranges>(__rngs)...);
+            return __parallel_transform_reduce_small_impl<512, 16, _Tp>(::std::forward<_ExecutionPolicy>(__exec), __n,
+                                                                        __reduce_op, __transform_op, __init,
+                                                                        ::std::forward<_Ranges>(__rngs)...);
         }
         else if (__n <= 16384)
         {
-            return __parallel_transform_reduce_small_submitter<512, 32, _Tp, _ReduceOp, _TransformOp>::submit(
-                ::std::forward<_ExecutionPolicy>(__exec), __n, __reduce_op, __transform_op, __init,
-                ::std::forward<_Ranges>(__rngs)...);
+            return __parallel_transform_reduce_small_impl<512, 32, _Tp>(::std::forward<_ExecutionPolicy>(__exec), __n,
```

Comment:
We rely on a unique pair of these compile time constants  in such calls.
So, it it unreliable for further code modification..

Reply:
I guess it's reliable because we should get immediate compile time error if we duplicate those and explicit name is passed